### PR TITLE
Final Phase Week 2

### PR DESCRIPTION
**Pull request by mrinalwahal**
Following is done:

1. Micro-service is now automated with `PyYaml` - as req in #19.
2. Supports two diff pipelines (one for snakemake and one custom) - as req in #23 
3. Front-end now only created `.drone.yml` - as req in #22 
4. #18 could have been closed last week, since the bot passed the checks then itself.
5. Keys are now ensured upon every login. If they don't exist, then fresh pair is generated and installed for the user. Tested and automated git ops for this with `GIT_SSH_COMMAND`.

Following is not done:

1. Drone secrets are not yet installed from `gin-proc`. This is a quick task for me, however, I'm facing a logical error - Drone only allows us to install secrets per repo. And in `gin-proc` we are taking the repo from user's during workflow submission, which is the time whence I'm no longer taking auth credentials from user. In this case, either I've have to hard-save the user's credentials in the code for repeatedly using everytime the user pushes to a new repo in `gin-proc` or I'll have to re-authenticate the user every time xe runs the workflow on a repo they haven't done before - this is to allow me to install the keys for that repo in Drone at the same time.

Following are some errors in the code:

1. Drone config commands is being appended instead of being over-written. I'm trying to cure this.

(I'm not sure why am I getting the entire commit history in this branch once again even after curing this issues last time. This was supposed to be a vanilla state branch. And that is why the bot is also failing its checks. I'm trying to solve the merge conflicts.)

### CODE REVIEW

**Comment by achilleas-k** [comment]
Looking into it now.  Can you check the merge conflicts in the meantime?

**Comment by achilleas-k** [comment]
Also noticed that when doing the push for the new drone.yml (see below), if any of the steps fail, it runs through and doesn't log the error.  In fact, it will print success.
https://github.com/G-Node/gin-proc/blob/3d30ee9c98effe491bac19e4dc96a92299760866/back-end/service.py#L262

File: back-end/service.py [hunk 1]
```diff
@@ -18,12 +20,15 @@
 
 # print(os.environ['GIT_SSH_COMMAND'])
 # print(os.environ['GIN_SERVER'])
-path = "http://" + os.environ['GIN_SERVER']
+GIN_ADDR = "http://" + os.environ['GIN_SERVER']
```
[the remark this answers is not in the document]
Comment (mrinalwahal) [reply back-end/service.py]: Should we make the user export env var for GIN_SERVER = `https://X.X.X.X:0000` ? I don't mind changing it right away.

[the remark this answers is not in the document]
Comment (achilleas-k) [reply back-end/service.py]: Yes, let's do it that way.

File: back-end/config.py [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+        commands.append("git annex sync --content")
+
+    return commands
+
+
+def createWorkflow(workflow, commands, user_commands=None):
+
+    if workflow == 'snakemake':
+        commands.append('snakemake')
+        commands.append('echo ".snakemake/" > .gitignore')
```
[the remark this answers is not in the document]
Comment (mrinalwahal) [reply back-end/config.py]: Alright. I'll try commenting out this line to see how it works out.

File: back-end/service.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrinalwahal) [reply back-end/service.py]: Done and pushed.

File: back-end/server.py [hunk 3]
```diff
@@ -35,8 +35,9 @@ def login():
         global token
         token = ensureToken(username, password)
         print('token ensured: {}'.format(token))
-        # print('key ensured: {}'.format(ensureKeys(token, auth.username)))
-        return ({'token': token}, 200)
+
+        if ensureKeys(token):
```
[the remark this answers is not in the document]
Comment (mrinalwahal) [reply back-end/server.py]: It sends an abort req to the front-end and that is supposed to show an error (or a red coloured notification like something, just like the blue-coloured success message you get on a 200 res. to check your drone window.

Updated code in latest push. You may check.

[the remark this answers is not in the document]
Comment (achilleas-k) [reply back-end/server.py]: Ah, good.

**Pull request merged**